### DISCUSSION TITLE
underscore_attrs_are_private

### DISCUSSION THREAD

**Discussion by omidekz** [pydantic/pydantic#2963] [Question]
seems config not work.
```
import pydantic as p
class Test(p.BaseModel):
	_name: str

	@property
	def name(self) -> str:
		return self._name

	class Config:
		underscore_attrs_are_private = False


a = Test(_name='test')
print(a.name)
```
this code raise error(Test obj has not attr _name)
what can i do for similar senarios?

**Comment by phemmer** [comment]
Private attributes can't be passed to the constructor. If you could, that'd mean they're public. They can only be set by operating on the instance attribute itself (e.g. `a._name = "foo"`).
